### PR TITLE
cleanup in Mesa easyblock + add custom sanity check step

### PR DESCRIPTION
**Pull request by boegel**
@edmondac Some additional changes for https://github.com/easybuilders/easybuild-easyblocks/pull/1892 (it got a little bit out of hand, but it all makes sense imho...)

### CODE REVIEW

File: easybuild/easyblocks/m/mesa.py [hunk 1]
```diff
@@ -26,17 +26,45 @@
...
+                feat_to_swrarch = {
+                    'avx': 'avx',
+                    'avx1.0': 'avx',  # on macOS, AVX is indicated with 'avx1.0' rather than 'avx'
+                    'avx2': 'avx2',
+                    'avx512f': 'skx',  # AVX-512 Foundation - introduced in Skylake
+                    'avx512er': 'knl',  # AVX-512 Exponential and Reciprocal Instructions implemented in Knights Landing
+                }
+                # determine list of values to pass to swr-arches configuration option
+                cpu_features = get_cpu_features()
+                self.swr_arches = sorted([feat_to_swrarch[key] for key in feat_to_swrarch if key in cpu_features])
```
Comment (Flamefire) [inline easybuild/easyblocks/m/mesa.py]: Why not "swrarch for feat, swrarch in feat_to_swrarch.items()..." as it was? More descriptive (and faster)

File: easybuild/easyblocks/m/mesa.py [hunk 2]
```diff
@@ -45,16 +73,42 @@ def configure_step(self, cmd_prefix=''):
...
-            if swr_arches:
-                self.cfg.update('configopts', "-Dswr-arches=%s" % ','.join(swr_arches))
-
-        return super(EB_Mesa, self).configure_step(cmd_prefix=cmd_prefix)
+        if self.swr_arches:
+            self.cfg.update('configopts', '-Dswr-arches=' + ','.join(self.swr_arches))
+
+        return super(EB_Mesa, self).configure_step()
+
+    def install_step(self):
```
Comment (Flamefire) [inline easybuild/easyblocks/m/mesa.py:82]: Where is this coming from? internal header doesn't sound like something a user would need. Also why is it not installed if it is meant to?

The description sounds like it belongs to a library  not installed. Wouldn't that be bad if the header exist but the library does not?

Comment (boegel) [reply easybuild/easyblocks/m/mesa.py:82]: see https://github.com/easybuilders/easybuild-easyconfigs/pull/9129

It seems like this is no longer needed for the most recent `Mesa` versions (hence the `os.path.exists` check)

File: easybuild/easyblocks/m/mesa.py (hunk 1 above)
Comment (boegel) [reply easybuild/easyblocks/m/mesa.py]: well, ok... done in 61f5389d7

File: easybuild/easyblocks/m/mesa.py [hunk 3]
```diff
@@ -45,16 +73,42 @@ def configure_step(self, cmd_prefix=''):
...
+            copy_dir(src_inc_GL_internal, target_inc_GL_internal)
+            self.log.info("Copied %s to %s" % (src_inc_GL_internal, target_inc_GL_internal))
+
+    def sanity_check_step(self):
+        """Custom sanity check for Mesa."""
+
+        shlib_ext = get_shared_lib_ext()
+
+        gl_inc_files = ['glext.h', 'gl_mangle.h', 'glx.h', 'osmesa.h', 'gl.h', 'glxext.h', 'glx_mangle.h']
+        gles_inc_files = [('GLES', 'gl.h'), ('GLES2', 'gl2.h'), ('GLES3', 'gl3.h')]
```
Comment (lexming) [inline easybuild/easyblocks/m/mesa.py]: These sanity checks will fail with Mesa v20. There's been some changes in v20 and now most of these headers are *no longer* installed if building with `libglvnd` (as we do in EB). You can check the changes between v19 and v20 in `include/meson.build`. The list of headers I used in https://github.com/easybuilders/easybuild-easyconfigs/pull/10175 reflects what is installed in v20.

Comment (boegel) [reply easybuild/easyblocks/m/mesa.py]: @lexming fixed in 4255b238f

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 4]
```diff
@@ -703,7 +703,7 @@ def sanity_check_step(self, *args, **kwargs):
             pip_version = det_pip_version()
             if pip_version:
                 if LooseVersion(pip_version) >= LooseVersion('9.0.0'):
-                    run_cmd("pip check", trace=False)
+                    run_cmd("cd %s; pip check -vvv" % self.installdir, trace=False)
```
Comment (Flamefire) [inline easybuild/easyblocks/generic/pythonpackage.py]: This wasn't intended, was it?

Comment (boegel) [reply easybuild/easyblocks/generic/pythonpackage.py]: ugh, no, thx

Comment (boegel) [reply easybuild/easyblocks/generic/pythonpackage.py]: fixed with force push in 4255b238f

File: easybuild/easyblocks/m/mesa.py [hunk 5]
```diff
@@ -45,16 +74,49 @@ def configure_step(self, cmd_prefix=''):
...
+            src_inc_GL_internal = os.path.join(self.start_dir, 'include', 'GL', 'internal')
+            copy_dir(src_inc_GL_internal, target_inc_GL_internal)
+            self.log.info("Copied %s to %s" % (src_inc_GL_internal, target_inc_GL_internal))
+
+    def sanity_check_step(self):
+        """Custom sanity check for Mesa."""
+
+        shlib_ext = get_shared_lib_ext()
+
+        if LooseVersion(self.version) >= LooseVersion('20.0'):
```
Comment (Flamefire) [inline easybuild/easyblocks/m/mesa.py:101]: Just thinking: If this changes so considerably between versions, wouldn't this be better done in the EC?

Comment (boegel) [reply easybuild/easyblocks/m/mesa.py:101]: Time will tell.

Going forward we can always easily overrule the easyblock by including `sanity_check_paths` in the easyconfig file.

File: easybuild/easyblocks/m/mesa.py (hunk 3 above)
Comment (lexming) [reply easybuild/easyblocks/m/mesa.py]: It's good now :+1: 
With Mesa v20 using `-Dgallium-drivers='swrast,swr'  -Dswr-arches=avx,avx2,skx`, the sanity checks for 
>{'files': ['lib/libOSMesa.so', 'include/EGL/eglmesaext.h', 'include/EGL/eglextchromium.h', 'include/GL/osmesa.h', 'include/GL/internal/dri_interface.h', 'lib/libswrAVX.so', 'lib/libswrAVX2.so', 'lib/libswrSKX.so'], 'dirs': ['include/GL/internal']}

**Review by lexming — APPROVED** [review]
LGTM

File: easybuild/easyblocks/m/mesa.py (hunk 5 above)
Comment (Flamefire) [reply easybuild/easyblocks/m/mesa.py:101]: Ah they take precedence. Didn't know that, than :+1:

**Pull request merged**